src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalysisState.cs

```diff
@@ -479,7 +480,7 @@ public async Task OnCompilationEventProcessedAsync(CompilationEvent compilationE
             }
 
             // Remove the event from event map.
-            lock (_gate)
+            using (_gate.DisposableWait())
```

Comment:
caller of this method do have cancellation

http://source.roslyn.io/#Microsoft.CodeAnalysis/DiagnosticAnalyzer/AnalyzerDriver.cs,961

can we propagate the cancellation here as well?

Reply:
tagging @mavasani

Reply:
This is updating state on compilation event processed, so I prefer to keep this one non-cancellable for now.

Reply:
In cases like that we should place a comment, and explicitly pass CancellationToken.None.  This lets readers in the future know that this behavior is intentional and was an accidental omission.  Stuff like that is *critical* for locking code as it can be so hard to get right, and so easy for people to screw this up in the future unintentionally.

Reply:
Done: https://github.com/dotnet/roslyn/pull/30593/commits/056c512107de6f18cc6f847cf93124ca8364010a